Uno.Gallery/Uno.Gallery.UWP/Views/Extensions/FolderSavePickerDialog.xaml.cs

```diff
@@ -0,0 +1,190 @@
+		private const string CancelResource = "FolderSavePickerCancel";
+		private const string TitleResource = "FolderSavePickerTitle";
+		private const string FileNameResource = "FolderSavePickerFileName";
+		private const string PickFolderResource = "FolderSavePickerPickFolder";
+		private const string BrowseResource = "FolderSavePickerBrowse";
+		private const string FileTypeResource = "FolderSavePickerFileType";
+
+		private static Lazy<ResourceLoader> _resourceLoader = new Lazy<ResourceLoader>(() => ResourceLoader.GetForViewIndependentUse());
+		private StorageFolder _pickedFolder;
+		private string[] _existingFileNames = Array.Empty<string>();
+
+		public FolderSavePickerDialog()
+		{
+			this.InitializeComponent();
+
+			Title = _resourceLoader.Value.GetString(TitleResource);
+			DefaultButton = ContentDialogButton.Primary;
+			PrimaryButtonText = _resourceLoader.Value.GetString(SaveResource);
+			SecondaryButtonText = _resourceLoader.Value.GetString(CancelResource);
+			FileNameSuggestBox.Header = _resourceLoader.Value.GetString(FileNameResource);
+			PickButton.Content = _resourceLoader.Value.GetString(BrowseResource);
+			PickFolderLabel.Text = _resourceLoader.Value.GetString(PickFolderResource);
+			FileTypeComboBox.Header = _resourceLoader.Value.GetString(FileTypeResource);
+
+			Loaded += FolderSavePickerDialog_Loaded;
+		}
+
+		private void FolderSavePickerDialog_Loaded(object sender, RoutedEventArgs e)
+		{
+			UpdatePickedFolderDisplay();
+		}
+
+		public string PickedFileName { get; private set; }
+
+		public StorageFolder PickedFolder
+		{
+			get => _pickedFolder;
+			private set
+			{
+				_pickedFolder = value;
+				UpdatePickedFolderDisplay();
+			}
+		}
+
+		public string SuggestedFileName
+		{
+			get => FileNameSuggestBox.Text;
+			set => FileNameSuggestBox.Text = value;
+		}
+
+		public void SetFileTypeChoices(IDictionary<string, IList<string>> fileTypeChoices)
+		{
+			var items = new List<FileTypeChoiceListItem>();
+			foreach (var choice in fileTypeChoices)
+			{
+				var extensions = string.Join(",", choice.Value.Select(extension => $"*{extension}"));
+				var label = $"{choice.Key} ({extensions})";
+				items.Add(new FileTypeChoiceListItem(label, choice.Value));
+			}
+			FileTypeComboBox.ItemsSource = items;
+			FileTypeComboBox.SelectedItem = items.FirstOrDefault();
+		}
+
+		private void SaveButtonClick(ContentDialog sender, ContentDialogButtonClickEventArgs args)
+		{
+			if (PickedFolder == null)
+			{
+				return;
+			}
+
+			IList<string> allowedExtensions = Array.Empty<string>();
+			if (FileTypeComboBox.SelectedItem is FileTypeChoiceListItem selectedFileTypeChoice)
+			{
+				allowedExtensions = selectedFileTypeChoice.Extensions;
+			}
+
+			bool needsExtension = false;
+			var fileName = FileNameSuggestBox.Text;
+			if (Path.HasExtension(fileName))
+			{
+				var extension = Path.GetExtension(fileName);
+				if (!allowedExtensions.Any(e => e.EndsWith(extension, StringComparison.InvariantCultureIgnoreCase)))

```

Comment:
whats is the format of `allowedExtensions`? `*.asd` or `.asd`
using EndsWith can cause issue with iso and so files